Update mod to release-20230225

penev92 opened the pull request:
The time has come around for the spiritual successor of #778.

To ease the review as much as possible this uses the latest ModSDK version from https://github.com/OpenRA/OpenRAModSDK/pull/183, the same as TDHD's `release-20230225` uses.

The last commit still needs work - chrome changes are not my strong side. Everything except that should be reviewable. Help with the UI is welcome.

Closes #580.

abcdefg30 commented on .vscode/tasks.json:
```diff
@@ -4,10 +4,34 @@
+		{
+			"id": "modId",
+			"description": "ID of the mod to run",
+			"default": "all",
```
```suggestion
			"default": "ra2",
```
(Commit: "Updated VSCode configuration")

abcdefg30 commented:
Commit "Updated ModSDK scripts from "release-20210321" to "release-20230225": Missing https://github.com/OpenRA/OpenRAModSDK/pull/185

abcdefg30 commented on mods/ra2/rules/world.yaml:
```diff
@@ -367,12 +365,36 @@ World:
+	HierarchicalPathFinderOverlay:
+	ColorPickerManager:
+		PreviewActor: amcv.colorpicker
+		PresetHues: 0, 0.125, 0.185, 0.4, 0.54, 0.66, 0.79, 0.875, 0, 0.14, 0.23, 0.43, 0.54, 0.625, 0.77, 0.85 # Copied from RA.
+		PresetSaturations: 0.9, 0.9, 0.9, 0.9, 0.9, 0.9, 0.9, 0.9, 0.4, 0.5, 0.4, 0.5, 0.4, 0.5, 0.4, 0.5 # Copied from RA.
```
The comments here can be removed I guess.

penev92 replied:
As you wish. I figured it may be good for traceability so someone doesn't wonder how the hell these numbers came to be in a year or two.

penev92 commented:
Fixed the observer/spectator UI chrome by replacing the newly-added `ingame-observer.yaml` in commit "**Manual mod updates PART 1 - mod.yaml**" with the correct one (was a copy of RA's, now is a copy of the removed Common one), so removed the last commit - "**SEND HELP - Observer/Spectator UI chrome**" as the chrome definitions don't need adjustment after all.

This should now be complete and ready for a full review.

penev92 commented:
Added voxel vehicle tilting on slopes as a bonus commit.

MustaphaTR commented:
Did a quick test, i noticed several issues (dropdowns have no separator, paradropping GIs are rendering normal body in addition to paradrop frame, deployed GIs do not turn, Soviet ConYard do not have damaged frames) but none seem to be related to/regressions from the update. One thing i think this PR could additionally have is applying https://github.com/OpenRA/OpenRA/pull/17783 but that could stay for another PR too.

About the ToDos you left on chrome.yaml you can copy those from RV, they are already properly updated there: https://github.com/MustaphaTR/Romanovs-Vengeance/blob/master/mods/rv/chrome.yaml

penev92 commented:
- Copied the chrome definitions that had newly-added TODOs from RV.
 - Added a "Bonus" commit to fix the missing dropdown separators, which had a TODO from #778. (credited that to @MustaphaTR as he's the one who came up with the coordinates).
 - Added a "Bonus" commit to add `InfiltrateForSupportPowerReset` (https://github.com/OpenRA/OpenRA/pull/17783 / https://github.com/OpenRA/ra2/issues/580).

There are more TODOs left over from #778 and things that aren't this PR's concern so it would probably be best not to address them here.

penev92 commented:
Added another small commit to fix content installer paths.

Mailaender commented:
Please clear https://github.com/OpenRA/ra2/pulls before merging this.
